src/stores/ContourConfigStore.ts

```diff
@@ -1,87 +1,91 @@
 import {action, computed, observable} from "mobx";
-import {RGBColor} from "react-color";
+import {CARTA} from "carta-protobuf";
 import {PreferenceStore} from "./PreferenceStore";
-import {hexStringToRgba} from "../utilities";
+import {hexStringToRgba, RGBA} from "../utilities";
+
+export enum ContourDashMode {
+    None,
+    Dashed,
+    NegativeOnly
+}
 
 export class ContourConfigStore {
     @observable enabled: boolean;
-    @observable numComputedLevels: number;
-    @observable lowerBound: number;
-    @observable upperBound: number;
-    @observable color: RGBColor;
+    @observable levels: number[];
+    @observable smoothingMode: CARTA.SmoothingMode;
+    @observable smoothingFactor: number;
+
+    @observable color: RGBA;
     @observable colormapEnabled: boolean;
     @observable colormap: string;
-    @observable dashLength: number;
-    @observable manualLevelsEnabled: boolean;
-    @observable manualLevels: number[];
+    @observable colormapContrast: number;
+    @observable colormapBias: number;
+    @observable dashMode: ContourDashMode;
+    @observable thickness: number;
 
     private readonly preferenceStore: PreferenceStore;
 
-    // Returns computed or manual contour levels
-    @computed get levels(): number[] {
-        // Default to manual levels if they are enabled
-        if (this.manualLevelsEnabled) {
-            return this.manualLevels;
-        } else if (isFinite(this.lowerBound) && isFinite(this.upperBound) && this.lowerBound < this.upperBound && this.numComputedLevels >= 1) {
-            // For single contour levels, just use the midpoint
-            if (this.numComputedLevels < 2) {
-                return [(this.upperBound + this.lowerBound) / 2.0];
-            } else {
-                // Fill in the steps linearly
-                const stepSize = (this.upperBound - this.lowerBound) / (this.numComputedLevels - 1);
-                const levelArray = new Array<number>(this.numComputedLevels);
-                for (let i = 0; i < levelArray.length; i++) {
-                    levelArray[i] = this.upperBound - i * stepSize;
-                }
-                return levelArray;
-            }
-        }
-
-        return [];
-    }
+    // // Returns computed or manual contour levels
```

Comment:
Are those commented code for future use?

Reply:
good catch :) I've removed it